specification/ai/OpenAI.Assistants/streaming/events.tsp

```diff
@@ -0,0 +1,163 @@
+import "@typespec/versioning";
+
+using TypeSpec.Versioning;
+
+namespace Azure.AI.OpenAI.Assistants;
+
+@doc("""
+Each event in a server-sent events stream has an `event` and `data` property:
+
+```
+event: thread.created
+data: {"id": "thread_123", "object": "thread", ...}
+```
+
+We emit events whenever a new object is created, transitions to a new state, or is being
+streamed in parts (deltas). For example, we emit `thread.run.created` when a new run
+is created, `thread.run.completed` when a run completes, and so on. When an Assistant chooses
+to create a message during a run, we emit a `thread.message.created event`, a
+`thread.message.in_progress` event, many `thread.message.delta` events, and finally a
+`thread.message.completed` event.
+
+We may add additional events over time, so we recommend handling unknown events gracefully
+in your code.
+""")
+union AssistantStreamEvent {
```

Comment:
I'm missing `string` as an option here?

Reply:
This `union` is a union of unions, so I thought that adding `string` wouldn't be correct. All the unions that are part of this `union` are basically string enums, so if the correct thing to do is to add `string`, I am more than happy to do that.

Reply:
I added `string` and the resulting generated OpenAPIv3 yaml changed only slightly, but it's actually more accurate:

![image](https://github.com/Azure/azure-rest-api-specs/assets/11056031/49ea15e3-c855-45f2-b55c-7b23350d4ff2)